### ISSUE TITLE
[CI] Adding MSVC build to CI check with Appveyor

### ISSUE THREAD

**Issue by NicolasDorier** [bitcoin/bitcoin#12613]
Credit to @sipsorcery for his hard work on making bitcoin compile in Visual Studio.

On top of this, he managed to integrate Visual Studio to integration server AppVeyor. (Travis does not support Windows for now as far as I know)

You can see it in action here: https://ci.appveyor.com/project/sipsorcery/bitcoin this is done on top of [this PR](https://github.com/bitcoin/bitcoin/pull/11526).

He shared the following config file for the CI:

```
version: 1.0.{build}
branches:
  only:
  - master
skip_tags: true
image: Visual Studio 2017
configuration: Release
platform: x64
environment:
  EMAIL: aaron@sipsorcery.com
  APPVEYOR_SAVE_CACHE_ON_ERROR: true
  DUMMY1: dummy
cache: C:\Tools\vcpkg\installed
before_build:
- ps: >-
    dir c:\tools\vcpkg

    dir c:\tools\vcpkg\installed


    vcpkg install boost:x64-windows-static

    vcpkg install libevent:x64-windows-static

    vcpkg install openssl:x64-windows-static

    vcpkg install zeromq:x64-windows-static

    vcpkg install berkeleydb:x64-windows-static

    vcpkg install secp256k1:x64-windows-static

    vcpkg install leveldb:x64-windows-static


    git pull origin pull/11526/head
     Invoke-WebRequest -Uri https://gist.githubusercontent.com/sipsorcery/0ce4b529ebd3bda784ad71c5702cae88/raw/6732cef26554fd016a154ed852f5dbac8fbf4199/bitcoin-config.h -OutFile src\config\bitcoin-config.h
build:
  project: build_msvc\bitcoin.sln
  parallel: true
  verbosity: minimal
artifacts:
- path: build_msvc\x64\Release\*.exe
  name: Executables
```

Note that the download of `bitcoin-config.h` will likely go away and be part of the build.

**Comment by sipsorcery** [comment]
Now slightly simplified with @NicolasDorier changes to VS project files.

`version: 1.0.{build}
branches:
  only:
  - master
skip_tags: true
image: Visual Studio 2017
configuration: Release
platform: x64
environment:
  EMAIL: aaron@sipsorcery.com
  APPVEYOR_SAVE_CACHE_ON_ERROR: true
  DUMMY1: dummy
cache: C:\Tools\vcpkg\installed
before_build:
- cmd: >-
    dir c:\tools\vcpkg

    dir c:\tools\vcpkg\installed


    vcpkg install boost:x64-windows-static

    vcpkg install libevent:x64-windows-static

    vcpkg install openssl:x64-windows-static

    vcpkg install zeromq:x64-windows-static

    vcpkg install berkeleydb:x64-windows-static

    vcpkg install secp256k1:x64-windows-static

    vcpkg install leveldb:x64-windows-static


    git pull origin pull/11526/head
build:
  project: build_msvc\bitcoin.sln
  parallel: true
  verbosity: minimal
artifacts:
- path: build_msvc\x64\Release\*.exe
  name: Executables`

**Comment by MarcoFalke** [comment]
It seems they are not running any tests at all

**Comment by NicolasDorier** [comment]
No, this would be done in a second step (as I saw you worked to make things runnable on windows)

Having it only compiling is already giving us good feedback on constructs which might break on windows though.
